### ISSUE TITLE
[feat] Automated accessibility check to our development pipeline.

### ISSUE THREAD

**Issue by AnnaGingle** [trussworks/react-uswds#2869]
### **Does your feature request relate to a specific USWDS component?**

No

### **What USWDS Version is this feature present in?**

Any.

### **Is your feature request related to a problem? Please describe.**

We should add an automated accessibility check to our development pipeline. 
- As government websites are required to meet section 508 compliance accessibility should be a north star rather than an afterthought. 
- Adding this check into the pipeline can help remediate one of the challenges when working on accessibility issues - finding out if the problem is tied to your configuration or the library

### **Describe the solution you'd like**

- Add an automated accessibility check to our development pipeline, specially section 508 / WCAG 2.0.
- Continuous Integration tests incorporate accessibility tools like axe and errors block commits.
- An automated process set up to optimize the use of developer time.

### **Describe alternatives you've considered**

On a previous project we used [Jest-axe](https://www.npmjs.com/package/jest-axe) (app testing)
[Storybook a11y add on](https://storybook.js.org/addons/@storybook/addon-a11y) (component level testing). @gidjin, @abbyoung, or @jcbcapps may be able to shed more light on that.

Well-known solutions to consider
- Deque's [aXe](https://www.deque.com/axe/) has changed the playing field for site reviews. Because of its flexible design, it can be executed from the command line, from within the browser, or even from a WYSIWYG editor. Being executed from the command line means that it is much easier to simply add accessibility checking to any scripts that are being run. It can be appended to a script crawling to ensure security best practices are maintained, or simply added to a continuous integration script.
- [Pa11y-ci ](https://github.com/pa11y/pa11y-ci) is a continuous integration (CI) runner that can test a few pages or an entire site for accessibility issues. 
   -  pa11y-ci in their CI pipelines and generate reports as artifacts or comments with the results of the testing.
  - Pa11y by default uses the [HTML_CodeSniffer](https://squizlabs.github.io/HTML_CodeSniffer/) accessibility engine, but it is recommended to switch to Pa11y with axe, which is [described in this UK Government Accessibility Manual](https://accessibility-manual.dwp.gov.uk/best-practice/automated-testing-using-axe-core-and-pa11y).
- [Cypress](https://www.cypress.io/) is a JavaScript testing framework that is used for automated testing. [Axe core](https://github.com/dequelabs/axe-core) is a popular accessibility testing engine.
  - Take a look at [Setting up Cypress with axe for accessibility](https://timdeschryver.dev/blog/setting-up-cypress-with-axe-for-accessibility) article that illustrates the setup.
  - [Cypress-axe](https://github.com/component-driven/cypress-axe)
- [Jest-axe](https://github.com/nickcolley/jest-axe) / Testing with [Jest](https://jestjs.io/) on GitHub
- [Nightwatch aXe-core](https://github.com/rikki-iki/nightwatch-axe-core) / Testing with [Nightwatch](https://nightwatchjs.org/) on GitHub

### **Additional context**
- [How I do automated accessibility testing for my website](https://opensource.com/article/23/2/automated-accessibility-testing) by Daniel Mundra

Most of this information is from the [CivicAction accessibility site](https://accessibility.civicactions.com/guide/tools). They put together some really impressive documentation on this topic.

**Comment by werdnanoslen** [comment]
Noting that I looked into [adding Playwright](https://github.com/trussworks/react-uswds/pull/2812), but that would only be useful locally (not as a GH-run action). We could have it run locally on commit, but I think we'd rather have it run on the repo as well to ensure no a11y bugs creep in.

**Comment by kimallen** [comment]
We already use @storybook/addon-a11y in development. @AnnaGingle are you recommending that we include it in the production version of storybook so that users looking at the deployed storybook instance can see the a11y scan results?

**Comment by kimallen** [comment]
We do use a linter, `eslint-plugin-jsx-a11y` in the development process. And the description suggests using Cypress/axe, though I only know of Cypress' use as an integration test tool. Since react-uswds is a library of components, @AnnaGingle how do you see Cypress being utilized in this case?

**Comment by werdnanoslen** [comment]
Ah, didn't realize Cypress was integration only — I imagined this feature would either run similar to the other Vitest/Jest unit tests (e.g. `expect(component).toBeAccessible`) or run on more complex stories without needing to be run locally ([see my Playwright experiment](https://github.com/trussworks/react-uswds/pull/2812))

**Comment by kimallen** [comment]
Recommendation from @JasonLin0991: https://www.deque.com/axe/core-documentation/api-documentation/
